PangleAdapter/src/main/java/com/chartboost/mediation/pangleadapter/PangleAdapter.kt

```diff
@@ -347,103 +403,72 @@ class PangleAdapter : PartnerAdapter {
     /**
      * Attempt to load a Pangle banner ad.
      *
-     * @param context The current [Context].
      * @param request An [PartnerAdLoadRequest] instance containing relevant data for the current ad load call.
      * @param partnerAdListener A [PartnerAdListener] to notify Chartboost Mediation of ad events.
      *
      * @return Result.success(PartnerAd) if the ad was successfully loaded, Result.failure(Exception) otherwise.
      */
     private suspend fun loadBannerAd(
-        context: Context,
         request: PartnerAdLoadRequest,
         partnerAdListener: PartnerAdListener
     ): Result<PartnerAd> {
         return suspendCoroutine { continuation ->
-            val bannerAd = TTAdSdk.getAdManager().createAdNative(context)
-
-            val adSlot = request.size?.let {
-                val widthDp = it.width.toFloat()
-                val heightDp = it.height.toFloat()
-
-                PartnerLogController.log(
-                    CUSTOM,
-                    "Pangle setting banner with size (w: $widthDp, h: $heightDp)"
-                )
-
-                AdSlot.Builder()
-                    .setCodeId(request.partnerPlacement)
-                    .setExpressViewAcceptedSize(
-                        widthDp,
-                        heightDp
-                    ).build()
-            }
-
-            bannerAd.loadBannerExpressAd(adSlot, object : TTAdNative.NativeExpressAdListener {
-                override fun onError(code: Int, message: String?) {
-                    PartnerLogController.log(LOAD_FAILED, "Code: $code. Error: $message")
-                    continuation.resume(
-                        Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN))
-                    )
-                }
+            PAGBannerAd.loadAd(
+                request.partnerPlacement,
+                PAGBannerRequest(getPangleBannerSize(request.size)),
+                object : PAGBannerAdLoadListener {
+                    override fun onError(code: Int, message: String?) {
+                        PartnerLogController.log(LOAD_FAILED, "Code: $code. Error: $message")
+                        continuation.resume(
+                            Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN))
+                        )
```

Comment:
We need a `resumeOnce()` implementation to be used in this method too.

Reply:
Oh, snap. 👀 Good call. Will update.